app/services/katello/candlepin/server.rb

```diff
@@ -0,0 +1,20 @@
+module Katello
+  module Candlepin
+    class Server
+      def self.init
+        return unless SETTINGS[:katello][:use_candlepin] && !Foreman.in_rake?('db:migrate')
+        candlepin_status = Katello::Ping.ping([:candlepin])
+
+        if candlepin_status[:status] == Katello::Ping::FAIL_RETURN_CODE
+          fail _("There was an issue with the Candlepin service %s: ") % candlepin_status[:candlepin][:message]
+        end
+
+        Organization.all.each do |org|
+          User.current = User.anonymous_admin
+          ForemanTasks.sync_task(::Actions::Katello::Organization::Create, org) unless org.library
```

Comment:
Maybe you can check if there are orgs without a library before running the ping. That would save a ping on already initialized instances.

I also wonder: foreman tasks are async so they should be able to run at a later time, so why is the ping even needed? At that later time candlepin might even be unreachable for whatever reason.